presto-main/src/main/java/com/facebook/presto/operator/ScanFilterAndProjectOperator.java

```diff
@@ -210,65 +209,63 @@ public Page getOutput()
             return null;
         }
 
-        if (!finishing) {
-            if ((pageSource == null) && (cursor == null)) {
-                ConnectorPageSource source = pageSourceProvider.createPageSource(operatorContext.getSession(), split, columns);
-                if (source instanceof RecordPageSource) {
-                    cursor = ((RecordPageSource) source).getCursor();
-                }
-                else {
-                    pageSource = source;
-                }
+        if (!finishing && pageSource == null && cursor == null) {
```

Comment:
The old code had a shared output structure that the new code doesn't need, but I like the idea of separate methods.